bin/docker-entrypoint.sh

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env bash
+# *****************************************************************************
+# This file is part of dirtsand.
+#
+# dirtsand is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# dirtsand is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with dirtsand.  If not, see <http://www.gnu.org/licenses/>.
+# *****************************************************************************
+
+DS_CONFIG_FILE=/opt/dirtsand/etc/dirtsand.ini
+CLIENT_CONFIG_FILE=/opt/dirtsand/etc/server.ini
+
+# Begin code copypasta'd from Deledrius
+# Check for existing configuration files and read existing keys if available, otherwise generate a new set
+echo -e "\e[36mPreparing game server...\e[0m"
+if [[ ! -f "$DS_CONFIG_FILE" || ! -f "$CLIENT_CONFIG_FILE" ]]; then
+    echo -e "\e[33mExisting configuration files not found or are incomplete.  Generating new configuration...\e[0m"
+    keyset=`$DS_BINARY --generate-keys`
+else
+    echo -e "\e[32mUsing existing configuration as a template.\e[0m"
+    keyset=`cat "$DS_CONFIG_FILE" "$CLIENT_CONFIG_FILE"`
+fi
+
+# Parse the keys we've either harvested or generated
+regex="([a-zA-Z]+\.[a-zA-Z]+\.[a-zA-Z])[ =\"]+([a-zA-Z0-9+=\/]+)\"*"
+while [[ $keyset =~ $regex ]]; do
+    keyname=${BASH_REMATCH[1]}
+    keyvalue=${BASH_REMATCH[2]}
+    case "$keyname" in
+        "Key.Auth.N")
+            kAuthN="$keyvalue" ;;
```

Comment:
Style note:  I usually prefer to put the `;;` on its own line in bash (similar to `break` in a C `switch`)